docs(k8s): add observability docs (#4290)
scaleway/docs-content: pages/kubernetes/how-to/monitor-data-plane-with-cockpit.mdx

1. **Promtail** can collect logs from:
   - **Container stdout/stderr** (pods)
   - **systemd journal** (e.g., `kubelet.service`)
2. **Log data** is transmitted to **Cockpit** (Loki).
3. **Cockpit** stores and indexes these logs.

## Step-by-step: Enabling container logs in Cockpit

You can use Scaleway’s **[Easy Deploy](/kubernetes/how-to/enable-easy-deploy/)** to add a Promtail deployment to your cluster:

1. **Open the Scaleway Console** and go to your **Kubernetes** cluster.
2. Navigate to the **Easy Deploy** tab.
3. Select **Promtail for Cockpit** from the library.
4. **Deploy** the application. Promtail will install on your cluster with default settings that:
   - Collect container logs for **all namespaces** (by default).
   - Collect systemd journal logs (e.g., `kubelet.service`).
   - Forward logs securely to **Cockpit**.
    <Message type="note">
      You can edit the default deployment configuration to filter logs by source (under `config.snippets.scrapeConfigs` in the YAML file). For example:
      ```yaml
      cockpit_promtail_scrape_config_pods: "namespace1,namespace2"
      cockpit_promtail_scrape_config_journal: "kubelet.service,kube-proxy.service"
      ```

>     </Message>

Reviewer: ```suggestion
    </Message>
    
```

### Example Promtail configuration
Below is a simplified snippet of the configuration that Easy Deploy generates by default:
```yaml
config:
  clients:
    - bearer_token: "{{{ cockpit_bearer_token }}}" # Automatically set by Easy Deploy
      url: "{{{ cockpit_loki_push_url }}}" # Automatically set by Easy Deploy

  snippets:
    scrapeConfigs: |
      {{{- cockpit_promtail_scrape_config_pods }}} # Default: log all pods
      {{{- cockpit_promtail_scrape_config_journal }}} # Default: log all system components
extraVolumeMounts:
  - mountPath: /var/log/journal
    name: journal
    readOnly: true
extraVolumes:
  - hostPath:
      path: /var/log/journal
    name: journal
```
<Message type="note">
  Template values like `{{{ cockpit_bearer_token }}}` (Bearer Token) and `{{{ cockpit_loki_push_url }}}` (Loki URL) are automatically set. Avoid modifying these values.
</Message>

> ## Observing logs in Cockpit

Reviewer: ```suggestion
## Visualizing logs in Cockpit
```

Once Promtail is running:

> 1. Go to **Cockpit** → **Kubernetes Cluster Pod Logs** (or open your own Grafana connected to Cockpit).

Reviewer: ```suggestion
1. Go to **Cockpit**, then **Kubernetes Cluster Pod Logs** (or open your own Grafana connected to Cockpit).
```

Reviewer: "Open your own Grafana connected to Cockpit" => do you mean the Grafana dashboards?

Author: yes

2. **Filter** by:
   - `Datasource` which is automatically created upon deployment, and visible in the Cockpit console
   - `Cluster Name` ( e.g. `my-kapsule-cluster`)
   - `namespace`, `pod`, or `container` labels to isolate specific workloads
   - **Time range** to limit how far back in history you want to query
3. **Analyze** logs in real-time or historical mode to troubleshoot issues, watch for errors, or track performance.

> ## Usage & pricing

Reviewer: ```suggestion
## Usage and pricing
```

> Sending logs to Cockpit is billed based on the **total volume of logs ingested**. You are charged *€0.15 per million log lines* (samples) or partial million.

Reviewer: ```suggestion
Sending logs to Cockpit is billed based on the **total volume of logs ingested**. You are charged *€0.35 per million log lines* (samples) or partial million.
```

Reviewer: Maybe we should just link to the Cockpit FAQ question about pricing in case pricing changes in the future?
https://www.scaleway.com/en/docs/faq/cockpit/#how-am-i-billed-for-using-cockpit-with-my-scaleway-data

---
meta:

>   title: How to monitor your Kubernetes Kapsule data plane with Cockpit

Author: It is a bit long, I think. I'll see if I can shorten it a bit

  description: This page explains how to integrate Kubernetes container logs with Scaleway Cockpit using Promtail
content:

>   h1: How to monitor your Kubernetes Kapsule Data Plane with Cockpit

Author: ```suggestion
  h1: How to monitor your Kubernetes Kapsule Data Plane with Cockpit
```
```suggestion
  h1: How to monitor your Kubernetes Kapsule data plane with Cockpit
```

This feature allows you to:

- **Enhance observability**: View logs from all your Kubernetes containers in one place.
- **Simplify troubleshooting**: Quickly drill down into specific pods or containers without needing to configure a separate logging stack.

<Message type="important">
    This feature does incur costs based on the volume of logs ingested. Refer to [Cockpit FAQ](/faq/cockpit/#how-am-i-billed-for-using-cockpit-with-custom-data) for more details and best practices to avoid unexpected bills.
</Message>

<Macro id="requirements" />

- A running [Kapsule](/kubernetes/how-to/create-cluster/) or [Kosmos](/kubernetes/how-to/create-kosmos-cluster/) cluster.
- An API Key with [IAM permissions](/iam/reference-content/permission-sets/) to edit your cluster `(KubernetesFullAccess` or `KubernetesSystemMastersGroupAccess`)

> - [A token](/cockpit/how-to/create-token/) with permissions to push to, and query logs from Cockpit

Author: Yes, the Kapsule team confirmed that no token is required but instead IAM permission.

## Architecture and limitations

### Control Plane vs. Data Plane

> - **Control Plane**: Fully managed by Scaleway. Users can already [monitor control plane components](/kubernetes/how-to/monitor-cluster/) (e.g., `kube-apiserver`, `CCM`, `CSI`) via Cockpit.  

Author: ```suggestion
- **Control plane**: Fully managed by Scaleway. Users can already [monitor control plane components](/kubernetes/how-to/monitor-cluster/) (e.g., `kube-apiserver`, `CCM`, `CSI`) via Cockpit.  
```

> - **Data Plane**: Runs in your Scaleway Project (customer-managed instances, `kubelet`, `containerd`, customer Pods, etc.). You have **full access** to the data plane, including the ability to SSH into nodes.

Author: ```suggestion
- **Data plane**: Runs in your Scaleway Project (customer-managed instances, `kubelet`, `containerd`, customer Pods, etc.). You have **full access** to the data plane, including the ability to SSH into nodes.
```

> | Feature            | Control Plane                                                                                                                                            | Data Plane                                                                                                        |

Author: ```suggestion
| Feature            | Control plane                                                                                                                                            | Data plane                                                                                                        |
```

      ```yaml
      cockpit_promtail_scrape_config_pods: "namespace1,namespace2"
      cockpit_promtail_scrape_config_journal: "kubelet.service,kube-proxy.service"
      ```
    </Message>
    
### Example Promtail configuration
Below is a simplified snippet of the configuration that Easy Deploy generates by default:
```yaml
config:
  clients:
    - bearer_token: "{{{ cockpit_bearer_token }}}" # Automatically set by Easy Deploy
      url: "{{{ cockpit_loki_push_url }}}" # Automatically set by Easy Deploy

  snippets:
    scrapeConfigs: |
      {{{- cockpit_promtail_scrape_config_pods }}} # Default: log all pods
      {{{- cockpit_promtail_scrape_config_journal }}} # Default: log all system components
extraVolumeMounts:
  - mountPath: /var/log/journal
    name: journal
    readOnly: true
extraVolumes:
  - hostPath:
      path: /var/log/journal
    name: journal
```
<Message type="note">
  Template values like `{{{ cockpit_bearer_token }}}` (Bearer Token) and `{{{ cockpit_loki_push_url }}}` (Loki URL) are automatically set. Avoid modifying these values.
</Message>

## Visualizing logs in Cockpit

Once Promtail is running:

> 1. Go to **Cockpit**, then **Kubernetes Cluster Pod Logs** (or open your own Grafana connected to Cockpit).

Reviewer 2: provide more details: 
1. Go to Cockpit, then Open dashboards.
2. In Grafana menu, go to dashboards and select **Kubernetes Cluster Pod Logs** (Note: need to be double-checked because there is currently a bug and it does not appear, so not sure, where it will be exactly)

2. **Filter** by:
   - `Datasource` which is automatically created upon deployment, and visible in the Cockpit console
   - `Cluster Name` ( e.g. `my-kapsule-cluster`)
   - `namespace`, `pod`, or `container` labels to isolate specific workloads
   - **Time range** to limit how far back in history you want to query
3. **Analyze** logs in real-time or historical mode to troubleshoot issues, watch for errors, or track performance.

## Usage and pricing

> Sending logs to Cockpit is billed based on the **total volume of logs ingested**. You are charged *€0.15 per million log lines* (samples) or partial million.

Author: ```suggestion
Sending logs to Cockpit is billed based on the **total volume of logs ingested**. Learn more about how you are billed for using Cockpit with Scaleway data [in the Cockpit FAQ](/faq/cockpit/#how-am-i-billed-for-using-cockpit-with-my-scaleway-data).
```

Key points include:

- **Logging rate**: The more logs you produce (e.g. high-traffic workloads or verbose logging), the higher the bill.
- **Filtering**: Limit logs to critical namespaces or system components only.

<Message type="note">
  You may edit the default configuration of the deployment to adjust the volume of logs to ingest:
  ```yaml
  extraLimitsConfig: |
    readline_rate_enabled: true # rate limiting
    readline_rate: 10000 # log lines / sec
    readline_burst: 10000 # cap for burst lines
    readline_rate_drop: true # drop excess lines
  ```

</Message>

<Message type="tip">

>   Always monitor the logs ingestion rate in the dedicated dashboards provided in Cockpit, just below the log tables, to avoid surprises.

Reviewer 2: would remove or clarify " just below the log tables, " in which Cockpit dashbaords ? Cockpit's overview or Kube's logs ?
